lib/bots/versionbot.ts

```diff
@@ -1550,6 +1571,157 @@ export class VersionBot extends ProcBot {
+	 * specified number of times for a PR.
+	 *
+	 * @param allCommits  All of the commits for the PR.
+	 * @param config      The config (if any) for the repository the PR belongs to.
+	 * @returns           An array of MissingTag objects, denoting required tags not on the PR.
+	 */
+	private checkCommitFooterTags(allCommits: GithubApiTypes.Commit[], config: VersionBotConfiguration): MissingTag[] {
+		const tagDefinitions = (config || {})['required-tags'] || [];
+		let tagCounts: { [key: string]: number } = {};
+		let changeTypeFound = false;
+
+		// Filter the list of commits to those not made by VB.
+		const commits = _.filter(allCommits, (commit) => commit.commit.committer.name !== process.env.VERSIONBOT_NAME);
+
+		// Get and validate tag configs.
+		for (let tag of tagDefinitions) {
+			// Lower the tag name.
+			tag.name = tag.name.toLowerCase();
+
+			if (tagCounts[tag.name]) {
+				throw new Error(`More than one occurrence of a required footer tag (${tag.name}) found ` +
+					'in configuration');
+			}
+			if (tag.occurrence) {
+				if ((typeof tag.occurrence !== 'number') && (tag.occurrence !== 'all')) {
+					throw new Error(`Invalid occurrence value found for ${tag.name} definition`);
+				}
+			}
+			// Always use a lowered version of the tag.
+			tagCounts[tag.name.toLowerCase()] = 0;
+
+			if (tag.name === 'change-type') {
+				changeTypeFound = true;
+			}
+		}
+
+		// Always add the 'change-type' tag.
+		if (!changeTypeFound) {
+			tagDefinitions.push({
+				name: 'change-type',
+				values: '/\s*(patch|minor|major)\s*$/i'
+			});
+			tagCounts['change-type'] = 0;
+		}
+
+		// Go through each commit. For each tag, we determine if it's present or not
+		// and if it matches required values.
+		for (let commit of commits) {
+			const commitMessage: string = commit.commit.message;
+
+			// Split the commits up into lines, and find the last line with any whitespace in.
+			// Whilst we tend to ask for:
+			//  <header>
+			//
+			//  <body>
+			//
+			//  <footer>
+			// This code will actually let you get away with:
+			//  <header>
+			//
+			//  <footer>
+			// As sometimes a development patch may be self-explanatory in the header alone.
+			const lines = commitMessage.split('\n');
+			const lastLine = _.findLastIndex(lines, (line) => line.match(/^\s*$/) );
+
+			// If there's no match, then at the very least there's no footer, and the commit
+			// is in the wrong format (as there's no text to use in the logs).
+			if (lastLine > 0) {
+				// We should have a line index to join from, now.
+				lines.splice(0, lastLine);
+				const footer = lines.join('\n');
+
+				// For each tag, interrogate the footer and determine if the tag is
+				// present.
+				// We check for a valid instance of a tag. If there are duplicates,
+				// potentially with invalid values, we don't flag it. This is primarily
+				// as we don't know how scripts requiring these tags deal with them
+				// and have to assume they'll pick a valid tag out.
```

Comment:
This bit of code hasn't changed. The `repository.yml` specifics an object for tags now, which means more than once instance of the key can't occur.
The search code searches for all occurrences of the tag name (of which there could be multiple in a commit) and then checks the value for each of those with a valid value ***and breaks out of the loop as soon as it finds one***. This inherently means you'll never get more than one positive result for a commit, regardless of the number of times the tag occurs, ensuring that the final count for all commits is true.